projects/js-toolkit/packages/liferay-cli/src/new/util/HumanError.ts

```diff
@@ -0,0 +1,10 @@
+/**
+ * SPDX-FileCopyrightText: © 2017 Liferay, Inc. <https://liferay.com>
+ * SPDX-License-Identifier: LGPL-3.0-or-later
+ */
+
+export default class HumanError extends Error {
+	constructor(message) {
+		super(message);
+	}
+}
```

Comment:
Whats the purpose of this class instead of using just Error?

Reply:
`HumanError` is used to distinguish build failure errors (things that failed because the user did something wrong which he needs to fix) from unhandled errors (our fault when programming the tool).

So, in the global `catch` I can check for `HumanError` vs `Error` and, when it is a `HumanError` show a pretty error message to the user hinting him what to do, whereas when it is a plain `Error` I show the stack trace of the code and blame it on use, so that the user can file an issue in our repo :roll_eyes: .

If it not clear, we may give it a more meaningful name, perhaps... `BuildError` or something like that?

Reply:
I probably meant `HumanReadableError` when I came up with that name. But I admit the naming was a bit unfortunate :-(